pjsip/include/pjsip-ua/sip_siprec.h

```diff
@@ -0,0 +1,135 @@
+#ifndef __PJSIP_SIPREC_H__
+#define __PJSIP_SIPREC_H__
+
+/**
+ * @file sip_siprec.h
+ * @brief SIP Session Recording Protocol (siprec) support (RFC 7866 - Session Recording Protocol in SIP)
+ */
+
+
+#include <pjsip-ua/sip_inv.h>
+#include <pjsip/sip_msg.h>
+
+/**
+ * @defgroup PJSIP_SIPREC SIP Session Recording Protocol (siprec) support (RFC 7866 - Session Recording Protocol in SIP)
+ * @brief SIP Session Recording Protocol support (RFC 7866 - Session Recording Protocol in SIP)
+ * @{
+ *
+ * \section PJSIP_SIPREC_REFERENCE References
+ *
+ * References:
+ *  - <A HREF="http://www.ietf.org/rfc/rfc7866.txt">RFC 7866: Session Recording Protocol (siprec)
+ *    in the Session Initiation Protocol (SIP)</A>
+ */
+PJ_BEGIN_DECL
+
+/**
+ * Initialize siprec module. This function must be called once during
+ * application initialization, to register siprec module to SIP endpoint.
+ *
+ * @param endpt         The SIP endpoint instance.
+ *
+ * @return              PJ_SUCCESS if module is successfully initialized.
+ */
+PJ_DECL(pj_status_t) pjsip_siprec_init_module(pjsip_endpoint *endpt);
+
+
+/**
+ * Create a=label attribute.
```

Comment:
Create or get?

Based on your implementation, I think you need both, but in separate APIs.

Get is to get the label, create is to add it in the response.